pkg/mcp/pods_exec_test.go

```diff
@@ -7,125 +7,132 @@ import (
-		mockServer := test.NewMockServer()
-		defer mockServer.Close()
-		c.withKubeConfig(mockServer.Config())
-		mockServer.Handle(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-			if req.URL.Path != "/api/v1/namespaces/default/pods/pod-to-exec/exec" {
-				return
-			}
-			var stdin, stdout bytes.Buffer
-			ctx, err := test.CreateHTTPStreams(w, req, &test.StreamOptions{
-				Stdin:  &stdin,
-				Stdout: &stdout,
-			})
-			if err != nil {
-				w.WriteHeader(http.StatusInternalServerError)
-				_, _ = w.Write([]byte(err.Error()))
-				return
-			}
-			defer func(conn io.Closer) { _ = conn.Close() }(ctx.Closer)
-			_, _ = io.WriteString(ctx.StdoutStream, "command:"+strings.Join(req.URL.Query()["command"], " ")+"\n")
-			_, _ = io.WriteString(ctx.StdoutStream, "container:"+strings.Join(req.URL.Query()["container"], " ")+"\n")
-		}))
-		mockServer.Handle(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-			if req.URL.Path != "/api/v1/namespaces/default/pods/pod-to-exec" {
-				return
-			}
-			test.WriteObject(w, &v1.Pod{
-				ObjectMeta: metav1.ObjectMeta{
-					Namespace: "default",
-					Name:      "pod-to-exec",
-				},
-				Spec: v1.PodSpec{Containers: []v1.Container{{Name: "container-to-exec"}}},
-			})
-		}))
-		podsExecNilNamespace, err := c.callTool("pods_exec", map[string]interface{}{
+type PodsExecSuite struct {
+	BaseMcpSuite
+	mockServer *test.MockServer
+}
+
+func (s *PodsExecSuite) SetupTest() {
+	s.BaseMcpSuite.SetupTest()
+	s.mockServer = test.NewMockServer()
+	s.Cfg.KubeConfig = s.mockServer.KubeconfigFile(s.T())
+}
+
+func (s *PodsExecSuite) TearDownTest() {
+	s.BaseMcpSuite.TearDownTest()
+	if s.mockServer != nil {
+		s.mockServer.Close()
+	}
+}
+
+func (s *PodsExecSuite) TestPodsExec() {
+	s.mockServer.Handle(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		if req.URL.Path != "/api/v1/namespaces/default/pods/pod-to-exec/exec" {
+			return
+		}
+		var stdin, stdout bytes.Buffer
+		ctx, err := test.CreateHTTPStreams(w, req, &test.StreamOptions{
+			Stdin:  &stdin,
+			Stdout: &stdout,
+		})
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			_, _ = w.Write([]byte(err.Error()))
+			return
+		}
+		defer func(conn io.Closer) { _ = conn.Close() }(ctx.Closer)
+		_, _ = io.WriteString(ctx.StdoutStream, "command:"+strings.Join(req.URL.Query()["command"], " ")+"\n")
+		_, _ = io.WriteString(ctx.StdoutStream, "container:"+strings.Join(req.URL.Query()["container"], " ")+"\n")
+	}))
+	s.mockServer.Handle(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		if req.URL.Path != "/api/v1/namespaces/default/pods/pod-to-exec" {
+			return
+		}
+		test.WriteObject(w, &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "default",
+				Name:      "pod-to-exec",
+			},
+			Spec: v1.PodSpec{Containers: []v1.Container{{Name: "container-to-exec"}}},
+		})
+	}))
+	s.InitMcpClient()
+
+	s.Run("pods_exec(name=pod-to-exec, command=[ls -l])", func() {
```

Comment:
nit, can we call out that `namespace=""` here - I liked how explicit the old test was, especially since that is the key for why the tool call fails

Reply:
Looking now, I understand what's supposed to happen - thanks!